wurst/event/DamageModify.wurst

```diff
@@ -0,0 +1,327 @@
+/**
+	DamageModify
+	by Trokkin
+
+	API:
+
+	// Pretty much '(Damage d) -> ...'
+	interface ForDamageCallback
+		run(Damage d)
+
+	// these are called before the damage is dealt
+	// can be called multiple amount of times or not called at all
+	addDamageModifier(ForDamageCallback cb)
+	removeDamageModifier(ForDamageCallback cb)
+	
+	// these are called after the damage is dealt
+	// are called strictly once per damage instance
+	addDamageListener(ForDamageCallback cb)
+	removeDamageListener(ForDamageCallback cb)
+
+	// a short for 'new Damage(this, target, amount, isSpell, id)..modify()..deal()'
+	unit.damage(unit target, real amount, OPTIONAL bool isSpell = true, int id = 0)
+
+	// a short for '= new Damage(this, target, amount, isSpell, id)..saveInstance()..modify()..deal()'
+	unit.returnDamage(unit target, real amount, bool isSpell, int id) -> Damage
+
+	class Damage
+		
+		// Basic damage event info
+		unit source
+		unit target
+		bool spell
+		real initial
+
+		// .id is used for any unique iterations
+		integer id
+		// .amount is used to manipulate damage amount
+		// like, addDamageModifier(d -> d.amount += d.initial * 1.00) adds +100% multiplier onto damage
+		real amount
+
+		// if .saveTarget was called then .target will survive from this damage no matter what
+		saveTarget()
+
+		// after damage has happened the Damage object is destroyed if saveInstance() wasn't called
+		saveInstance()
+		// releaseInstance is used to destroy Damage objects. Use this instead of 'destroy'!
+		releaseInstance()
+
+		// Some obvious wrappers
+		addAmount(real amount)
+		addMultiplier(real mult)
+
+		// .modify() evaluates all damageModifiers on itself
+		// is automatically called on non-triggered damage once
+		// isn't called on manual construction, so user can manually decide how many iterations a damage should have, from 0 to infinity
+		modify()
+
+		// Returns true if the damage is going to kill or already have killed target
+		isKillingBlow() -> bool
+
+		// Damage objects can be created manually to have full control on them from the code
+		construct(unit source, unit target, real amount, bool isSpell, int id)
+		// .deal() runs manually created damage
+		deal()
+
+	// an important note:
+	// for damageListeners .amount is set to actual difference in target's life
+	// so damage won't be more than target's life if it has killed target (or it was saved)
+	// (a unit has 10 hp, it takes 100 damage, for damageModifiers it's 100 damage
+		but for damageListeners it's 10 damage / 10 - .405 damage if .doSaveTarget == true)
```

Comment:
missing `//` prefix, also prefer ", or" instead of `/` here which can be confused with division.

I think this important note probably should be in hotdoc comments instead, but not sure.